File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> ans(2);
        int si = 0;
        int ei = numbers.size()-1;
        
        while(si < ei){
            if(numbers[si] + numbers[ei] == target){
                ans[0] = si+1;
                ans[1] = ei+1;
                return ans;
            }else if(numbers[si] + numbers[ei] < target){
                si++;
            }else{
                ei--;
            }
        }
        
        return ans;
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers)-1
        res = [0, 0]
        
        while l<r:
            s = numbers[l]+numbers[r]
            if s ==target:
                res[0] = l+1
                res[1] = r+1
                break
            elif s<target:
                l+=1
            else:
                r -= 1
        
        return res